src/image/image_raw.rs

```diff
@@ -131,48 +135,119 @@ where
 {
     /// Creates a new image.
     ///
-    /// # Panics
-    ///
-    /// If `data` doesn't have the correct length.
-    pub fn new(data: &'a [u8], width: u32, height: u32) -> Self {
-        let ret = Self {
+    /// Only the width of the image needs to be specified. The height of the image will be
+    /// calculated based on the length of the given image data.
+    pub fn new(data: &'a [u8], width: u32) -> Self {
+        // Prevent panic for `width == 0` by returning a zero sized image.
+        if width == 0 {
+            return Self {
+                data: &[],
+                size: Size::zero(),
+                pixel_type: PhantomData,
+                byte_order: PhantomData,
+            };
+        }
+
+        let height = data.len() / bytes_per_row(width, C::Raw::BITS_PER_PIXEL);
```

Comment:
If a user gives data that would result in a partial final row, how should we handle this? For example, this test fails:

```rust
#[test]
fn partial_row() {
    let data = [
        0xAA, 0x00, //
        0x55, 0xFF, //
        0xAA, 0x80, //
        0xBB,
    ];
    let image_data: ImageRaw<BinaryColor> = ImageRaw::new(&data, 9);

    assert_eq!(image_data.size(), Size::new(9, 4));
}
```

With a height of `3` but if I got my logic right it should indeed have a height of `4` (if we want to handle partial rows). If the desired behaviour is to ignore that final row, we should mention that in the docs I think.

Reply:
Good point. I'm not entirely sure what the best behavior is in this case. I chose to ignore any partial rows, because it would make sure that the iterator always outputs a full rectangle. But perhaps drawing an incomplete image would be easier to debug for the user.

I've added a note to describe the current behavior. But if you like I could also try to change it to draw incomplete rows instead.

Reply:
I'm also unsure, so let's leave it as is with the note. I think it's easier to "fix" this behaviour later in a point release than it would be to remove partial row rendering if for some reason that ends up being the wrong approach.

Reply:
OK. Thanks for the review.